ios/brave-ios/Sources/Brave/Frontend/Browser/Tabs/TabGrid/TabGridViewModel.swift

```diff
@@ -262,8 +262,14 @@ class TabGridViewModel {
 
   private func browsingModeDidSwitch(from oldValue: Bool) {
     tabManager.privateBrowsingManager.isPrivateBrowsing = isPrivateBrowsing
-    if isPrivateTabsLocked {
+    // Clear out the tabs if we are in private mode otherwise
+    // tabs from the non-private mode will be temporarily avaialable.
+    // Clearing out the tabs ensures that the shred menu button is hidden in an empty private mode
+    if isPrivateBrowsing {
       tabs = []
+    }
+    
+    if isPrivateTabsLocked {
       return
     }
 
```

Comment:
i can still reproduce. 
the updateTabs() below already updates `tabs` correctly. 
the problem the button is still visible is in the logic of how we compute `isShredMenuVisible`. we need to check if the current selected tab `in the current mode` is shred available